include/phasar/DataFlow/IfdsIde/Solver/JumpFunctions.h

```diff
@@ -157,6 +158,18 @@ template <typename AnalysisDomainTy, typename Container> class JumpFunctions {
     return NonEmptyLookupByTargetNode[Target];
   }
 
+  template <typename HandlerFn>
+  void foreachEdgeFunction(HandlerFn Handler) const {
+    NonEmptyForwardLookup.foreachCell(
+        [Handler = std::move(Handler)](ByConstRef<d_t> /*Row*/,
+                                       ByConstRef<n_t> /*Col*/,
+                                       const auto &TargetFactAndEF) {
+          for (const auto &[TargetFact, EF] : TargetFactAndEF) {
+            std::invoke(Handler, EF);
```

Comment:
If the callback is a template parameter, I use `std::invoke` to achieve maximum flexibility (allowing member pointers and member function pointers to be called, as well as "seeing through" `std::reference_wrapper`).

Otherwise, I use regular function-call syntax, which just looks nicer